dll/win32/shell32/wine/pidl.c

```diff
@@ -1722,6 +1722,80 @@ LPITEMIDLIST _ILCreateGuidFromStrA(LPCSTR szGUID)
 }
 #endif
 
+// Create GUID from shortcut(desktop, Personal...)
+// returns TRUE if succefull and FALSE otherwise
+BOOL _ILCreateGuidFromShortcutW(LPCWSTR szShortcut, GUID *fidOut)
+{
+	if (!StrCmpIW(szShortcut, L"ADMINISTRATIVE TOOLS"))
+	{
+		*fidOut = CLSID_AdminFolderShortcut;
+		return TRUE;
+	}
+	else if (!StrCmpIW(szShortcut, L"CONTROLPANELFOLDER"))
+	{
+		*fidOut = CLSID_ControlPanel;
+		return TRUE;
+	}
+	else if (!StrCmpIW(szShortcut, L"FONTS"))
+	{
+		*fidOut = CLSID_FontsFolderShortcut;
+		return TRUE;
+	}
+	else if (!StrCmpIW(szShortcut, L"MYCOMPUTERFOLDER"))
+	{
+		*fidOut = CLSID_MyComputer;
+		return TRUE;
+	}
+	else if (!StrCmpIW(szShortcut, L"PERSONAL"))
+	{
+		*fidOut = CLSID_MyDocuments;
+		return TRUE;
+	}
+	else if (!StrCmpIW(szShortcut, L"NETWORKPLACESFOLDER"))
+	{
+		*fidOut = CLSID_NetworkPlaces;
+		return TRUE;
+	}
+	else if (!StrCmpIW(szShortcut, L"CONNECTIONSFOLDER"))
+	{
+		*fidOut = CLSID_NetworkConnections;
+		return TRUE;
+	}
+	else if (!StrCmpIW(szShortcut, L"PRINTERSFOLDER"))
+	{
+		*fidOut = CLSID_Printers;
+		return TRUE;
+	}
+	else if (!StrCmpIW(szShortcut, L"RECYCLEBINFOLDER"))
+	{
+		*fidOut = CLSID_RecycleBin;
+		return TRUE;
+	}
+	return FALSE;
+}
+
+// if szUrl is '::{GUID}' then returns a GUID
+// else it tries to resolve shortcut
+LPITEMIDLIST _ILCreateGuidFromShellUrlW(LPCWSTR szUrl)
+{
+    GUID iid;
+
+	if (szUrl[0] == L':' && szUrl[1] == L':')
+	{
+		if (!GUIDFromStringW(szUrl + 2, &iid))
+		{
+			ERR("%s is not a GUID\n", debugstr_w(szUrl));
+			return NULL;
+		}
+	}
+	else if (!_ILCreateGuidFromShortcutW(szUrl, &iid))
+	{
+		ERR("%s is not a shell shortcut\n", debugstr_w(szUrl));
+		return NULL;
+	}
+    return _ILCreateGuid(PT_GUID, &iid);
+}
```

Comment:
Should those two functions placed in file of "wine" subdirectory?
At least they should be surrounded by a `#ifdef __REACTOS__  ...  #endif` guard, since they are not in Wine.